tests/components/imeon_inverter/test_config_flow.py

```diff
@@ -0,0 +1,157 @@
+    mock_async_setup_entry: Generator[AsyncMock],
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], TEST_USER_INPUT
+    )
+    await hass.async_block_till_done()
+
+    assert result2["type"] is FlowResultType.CREATE_ENTRY
+    assert result2["title"] == TEST_USER_INPUT[CONF_ADDRESS]
+    assert result2["data"] == TEST_USER_INPUT
+    assert mock_async_setup_entry.call_count == 1
+
+
+async def test_form_invalid_auth(
+    hass: HomeAssistant, mock_imeon_inverter: Generator[MagicMock]
+) -> None:
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    mock_imeon_inverter.login.return_value = False
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], TEST_USER_INPUT
+    )
+
+    assert result2["type"] is FlowResultType.FORM
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+    mock_imeon_inverter.login.return_value = True
+
+    result3 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], TEST_USER_INPUT
+    )
+
+    assert result3["type"] is FlowResultType.CREATE_ENTRY
+
+
+@pytest.mark.parametrize(
+    ("error", "expected"),
+    [
+        (TimeoutError, "cannot_connect"),
+        (ValueError("Host invalid"), "invalid_host"),
+        (ValueError("Route invalid"), "invalid_route"),
+        (ValueError, "unknown"),
+    ],
+)
+async def test_form_exception(
+    hass: HomeAssistant,
+    mock_imeon_inverter: Generator[MagicMock],
+    error: Exception,
+    expected: str,
+) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    mock_imeon_inverter.login.side_effect = error
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], TEST_USER_INPUT
+    )
+
+    assert result2["type"] is FlowResultType.FORM
+    assert result2["errors"] == {"base": expected}
+
+    mock_imeon_inverter.login.side_effect = None
+
+    result3 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], TEST_USER_INPUT
+    )
+
+    assert result3["type"] is FlowResultType.CREATE_ENTRY
+
+
+async def test_manual_setup_already_exists(
+    hass: HomeAssistant,
+    mock_async_setup_entry: Generator[AsyncMock],
+    mock_imeon_inverter: Generator[MagicMock],
+) -> None:
+    """Test that a flow with an existing host aborts."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=TEST_SERIAL,
+        data=TEST_USER_INPUT,
+    )
+
+    entry.add_to_hass(hass)
```

Comment:
```suggestion
    mock_imeon_inverter: Generator[MagicMock],
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test that a flow with an existing host aborts."""

    mock_config_entry.add_to_hass(hass)
```
we already have one in conftest we can reuse